recipes/rapidjson/config.yml

```diff
@@ -1,3 +1,5 @@
 versions:
   "1.1.0":
     folder: "1.1.x"
+  "20200410":
+    folder: "all"
```

Comment:
I am mixing tagged release with using master, it's been 4 years since the last release and 500+ commits. 

is this the correct approach? It's 99% copy paste however.

Reply:
related https://github.com/Tencent/rapidjson/issues/1006

Reply:
Yeah there's a ton of issues asking for one lol

Reply:
I don't like mixing at all.

How much work would it be to backport this one bug fix?

Reply:
I don't either...

A lot. I couldn't find an issue, and I'm not sure what code was broken in the first place. Updated my local copy to master and the crash went away. The issue and commit might have been written in chinses making it harder to track down.

Reply:
@prince-chrismc 
- Maybe you can try `git bisect` to find what change fixed your issue?

- Maybe add the date of `1.1.0` as a comment.
Something like:
```
versions:
  "1.1.0":  # 20160825
    folder: "1.1.x"
  "20200410":
    folder: "all"
```

Reply:
Maybe we should find a new naming schema for situations like that

Random idea: A `-next` suffix. So in this case `1.1.0-next`?

I would like this much better as it is clear where in the version history this version is located. Matching dates and versions makes this unclear

Reply:
That's a good idea, however what happens when someone needs the next upstream update?

Would `1.2.0-dev.0` work for you?

Reply:
short commit hash?

Reply:
> short commit hash?

Same problem as with mixing dates/version numbers.

When looking at 
`2.3.0` and `20200302`
or
`1.1.0-next-abcdfg` and `1.1.0-next-a1b2c3d`
you have no clue what is newer without additional informations

> Would 1.2.0-dev.0 work for you?

That would be **_guessing_** what the next version number is, so that is probably bad too

> That's a good idea, however what happens when someone needs the next upstream update?

Good point. Since your suggestion also contains a build number, we could combine that to `1.1.0-next.0`, `1.1.0-next.1` etc.
From my understanding this is also still valid [semver](https://semver.org)

What do you think?

Reply:
Or create a version format that lives as a subspace in the project version space?
Something like:
- `1.1.0.conan.20200410`
- `1.1.0.cci.20200410`

You know immediately that it is a conan version and if it is newer then an official release.

Reply:
I regret such mixing cannot be used. many people are using conan version ranges, and adding 20200410 as a version will break it. we already had similar issues in the past with glm.

Reply:
I suggest `next` as the keyword because I'm assuming that the chance is higher to understand that is newer that the version number in front of it. 

`1.1.0.conan.20200410` could be a Conan-special/patched build of the 1.1.0 version?

Reply:
> I regret such mixing cannot be used. many people are using conan version ranges, and adding 20200410 as a version will break it. we already had similar issues in the past with glm.

this will happen, and there is no way to deal with it. I added e.g. bacnet-stack to cci with e.g. `yyyymmdd` though this was made because the last release was 2 years ago and in the meantime the project moved to github, adapter cmake etc. 

the next release will be semver again...

Reply:
I think cci is perfect, it clearly denotes the 'version' is unique to CCI, the last segment being the date makes it manageable for a human to sort.

ended up with `1.1.0-cci.20200410` to respect the syntax requirements

Reply:
Does "-xxx" indicate a semver *pre*-release? Won't "1.1.0" be considered later than "1.1.0-cci.yyyymmdd"?

Reply:
> Does "-xxx" indicate a semver _pre_-release? Won't "1.1.0" be considered later than "1.1.0-cci.yyyymmdd"?

Yes and even in the code it says it less... that's a bit awkward :confused:

Reply:
Yes, semantically we are a bit off here, but I think that this should be okay. After all, this is already an exception, it shouldn't be up to packagers to create new releases in the first place. And personally, I think it is still better than to guess what the next version number might be

Reply:
IMHO, we should think about a version naming that won't be resolved using SemVer's version ranges. 
* If the consumer uses `rapidjson/[>=1.1.0]` this _unnofficial_ version should be taken into account (this invalidates my previous comment about using just the date).
* And we cannot use a `pre-release` in SemVer schema because this version is `1.1.0` with more changes 
* We don't know when the next tag of `rapidjson` will happen and which commit they will tag, so we can't know if this is related to `1.1.0` version or it should have been something like `1.2.0` plus changes, or these are changes after a hypothetical `2.x.y` version.

I feel like it is better to use something totally unrelated, outside SemVer resolution, maybe `cci.yyyymmdd`, it won't be resolved by version_ranges (I need to check how Conan behaves here). If you want to use this version, you need to write it explicitly in the consumer.

I know that using two different versioning schemas can be confusing, but for packages that don't follow semver (we don't know if any of those 500+ commits has broken the API) better not to use something that fits into semver's rules for version_ranges.

Please, wdyt from the consumer POV?

Reply:
> We don't know when the next tag of rapidjson will happen

I reached out to find out, it will be very unlikely that there will ever be a new release.

> we don't know if any of those 500+ commits has broken the API

There's a few API changes, I needed to modify my code

Reply:
We can not mix numeric release versions and a date version. [ref](https://github.com/conan-io/conan-center-index/pull/1464#discussion_r415741534) So we are stuck taking something that respects semVer...

After more digging into this project there is a [v1.2](https://github.com/Tencent/rapidjson/milestone/4) target which was presented

Since this is a "future release" could we then use `1.2.0-cci.20200410`

There was concerns about "guessing" the next release number but this is the _official_ word from the main developer done here [just after this comment](https://github.com/Tencent/rapidjson/issues/1006#issuecomment-395997972)

@jgsogo @Croydon 

Would `1.2.0-cci.20200410` be a suitable compromise?

Reply:
It would be somewhat okay-ish for me in this case, but I don't think that it is a solution that scales for all projects which might be in a similar situation.

While it looks pretty safe for rapidjson that the next version is 1.2, plans can change at any time and we can't really de-publish any version.

What if rapidjson (or any other recipe/project in a similar situation) decides that next version is actually named `1.1.1` after we published `1.2.0-cci.something`? People using version ranges would then get an older AND unofficial version rather than the latest official version.

I think that this is much, much worse than consciously breaking the semantic meaning of semver for an unofficial version, which should probably only be explicitly used anyway.

If we use `1.1.0-cci.20200410` the people using version ranges won't get this unofficial release automatically, but rather the official release `1.1.0`, which might be favourable.

Reply:
What about a separate projects?

- rapidjson/1.1.0
- rapidjson-master/20200410

There's no mixing and everything follows the current model, with a slight twist.

Reply:
Is using  build identifiers out of the question?
something like: `rapidjson/1.1.0+cci20200410`